admin/test/test_release.py

```diff
@@ -1842,40 +1844,191 @@ def test_tag_exists_fails(self):
             self.calculate_base_branch, '0.3.0')
 
 
-class CopyTutorialVagrantBox(SynchronousTestCase):
+class PublishVagrantMetadataTests(SynchronousTestCase):
     """
-    Tests for :func:`copy_tutorial_vagrant_box`.
+    Tests for :func:`publish_vagrant_metadata`.
     """
 
-    def test_vagrant_box_copied(self):
+    def setUp(self):
+        self.target_bucket = 'clusterhq-archive'
+        self.metadata_key = 'vagrant/flocker-tutorial.json'
+
+    def publish_vagrant_metadata(self, aws, version):
+        scratch_directory = FilePath(self.mktemp())
+        scratch_directory.makedirs()
+        box_url = "https://example.com/flocker-tutorial-{}.box".format(version)
+        box_name = 'flocker-tutorial'
+        sync_perform(
+            ComposedDispatcher([aws.get_dispatcher(), base_dispatcher]),
+            publish_vagrant_metadata(
+                version=version,
+                box_url=box_url,
+                box_name =box_name,
+                target_bucket=self.target_bucket,
+                scratch_directory=scratch_directory))
+
+    def test_no_metadata_exists(self):
         """
-        A Vagrant box for a given version of Flocker is copied to the
-        archive.
+        A metadata file is added when one does not exist.
         """
-        target_bucket = 'clusterhq-archive'
-        dev_bucket = 'clusterhq-dev-archive'
+        aws = FakeAWS(
+            routing_rules={},
+            s3_buckets={
+                self.target_bucket: {},
+            },
+        )
+
+        self.publish_vagrant_metadata(aws=aws, version='0.3.0')
+
+        expected_metadata = {
+            "description": "clusterhq/flocker-tutorial box.",
+            "name": "clusterhq/flocker-tutorial",
+            "versions": [
+                {
+                    "version": "0.3.0",
+                    "providers": [
+                        {
+                            "url": "https://example.com/flocker-tutorial-0.3.0.box",  # noqa
+                            "name": "virtualbox"
+                        }
+                    ],
+                }
+            ],
+        }
```

Comment:
It's a shame to repeat this somewhat large structure throughout the tests.  Only a couple parts of it vary in interesting ways.  I suggest defining a function that takes one or two parameters and returns the full thing and then using that instead.